crates/cext/src/dag.rs

```diff
@@ -157,6 +163,509 @@ pub unsafe extern "C" fn qk_dag_num_clbits(dag: *const DAGCircuit) -> u32 {
+///
+/// Panics if the node is not an operation.
+///
+/// @param dag A pointer to the DAG.
+/// @param node The operation node to get the number of qubits of.
+///
+/// @return The number of qubits of the operation.
+///
+/// # Safety
+///
+/// Behavior is undefined if ``dag`` is not a valid, non-null pointer to a ``QkDag``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_dag_op_node_num_qubits(dag: *const DAGCircuit, node: u32) -> u32 {
+    let dag = unsafe { const_ptr_as_ref(dag) };
+    let instr = dag
+        .dag()
+        .node_weight(NodeIndex::new(node as usize))
+        .expect("expected operation node")
+        .unwrap_operation();
+    instr.op.num_qubits()
+}
+
+/// @ingroup QkDag
+/// Gets the number of clbits of the specified operation node.
+///
+/// Panics if the node is not an operation.
+///
+/// @param dag A pointer to the DAG.
+/// @param node The operation node to get the number of clbits of.
+///
+/// @return The number of clbits of the operation.
+///
+/// # Safety
+///
+/// Behavior is undefined if ``dag`` is not a valid, non-null pointer to a ``QkDag``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_dag_op_node_num_clbits(dag: *const DAGCircuit, node: u32) -> u32 {
+    let dag = unsafe { const_ptr_as_ref(dag) };
+    let instr = dag
+        .dag()
+        .node_weight(NodeIndex::new(node as usize))
+        .expect("expected operation node")
+        .unwrap_operation();
+    instr.op.num_clbits()
+}
+
+/// @ingroup QkDag
+/// Gets the number of params of the specified operation node.
+///
+/// Panics if the node is not an operation.
+///
+/// @param dag A pointer to the DAG.
+/// @param node The operation node to get the number of params of.
+///
+/// @return The number of params of the operation.
+///
+/// # Safety
+///
+/// Behavior is undefined if ``dag`` is not a valid, non-null pointer to a ``QkDag``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_dag_op_node_num_params(dag: *const DAGCircuit, node: u32) -> u32 {
+    let dag = unsafe { const_ptr_as_ref(dag) };
+    let instr = dag
+        .dag()
+        .node_weight(NodeIndex::new(node as usize))
+        .expect("expected operation node")
+        .unwrap_operation();
+    instr.op.num_params()
+}
+
+/// @ingroup QkDag
+/// Retrieve the qubits of the specified operation node.
+///
+/// Panics if the node is not an operation.
+///
+/// @param dag A pointer to the DAG.
+/// @param node The operation node to get the qubits of.
+///
+/// @return A pointer to the qubits. Use ``qk_dag_op_node_num_qubits`` to determine the number of
+///     elements.
+///
+/// # Safety
+///
+/// Behavior is undefined if ``dag`` is not a valid, non-null pointer to a ``QkDag``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_dag_op_node_qubits(dag: *const DAGCircuit, node: u32) -> *const u32 {
+    let dag = unsafe { const_ptr_as_ref(dag) };
```

Comment:
We stopped doing safety comments on these guys a few functions back.  Maybe doesn't hurt _too_ much to copy-paste the
```rust
// SAFETY: per documentation, this points to a valid intiialised DAGCircuit
```
or whatever it is, if not only so we're habitual about it?

I don't feel desperately strongly.